cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -1393,23 +1419,76 @@ func (p *BicepProvider) loadParameters(ctx context.Context) (map[string]azure.Ar
 	return armParameters.Parameters, nil
 }
 
+type compiledBicepParamResult struct {
+	TemplateJson   string `json:"templateJson"`
+	ParametersJson string `json:"parametersJson"`
+}
+
+type compileBicepResult struct {
+	RawArmTemplate azure.RawArmTemplate
+	Template       azure.ArmTemplate
+	Parameters     azure.ArmParameters
+}
+
 func (p *BicepProvider) compileBicep(
 	ctx context.Context, modulePath string,
-) (azure.RawArmTemplate, azure.ArmTemplate, error) {
-	compiled, err := p.bicepCli.Build(ctx, modulePath)
+) (*compileBicepResult, error) {
```

Comment:
@ellismg , here's the most significant change.
`compileBicep()` will now check if the `modulePath` is a `bicepparam` or not.

This strategy allows anyone currently calling `compileBicep()` not to worry about if there is a `bicepparam` or not.  The result is returned now as a `compileBicepResult` (instead of a RawArmTemplate).

I used this strategy because azd is currently doing a double-bicep-compilation. The first time is only to check if user is trying to do a resource-group deployment and the second is for the actual `provisioning`.